### PR TITLE
[dbnode] Skip snapshot/flush if shard is not bootstrapped.

### PR DESCRIPTION
**Pull request by notbdu**
**What this PR does / why we need it**:

Skips snapshotting/flushing if shard is not bootstrapped.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

**Review by linasm — COMMENTED** [review]
Can we cover this fix from unit tests? I.e. return `false` from one of the mocked `shard.IsBootstrapped()` and assert that it gets skipped during the processing (in each of the flows affected)?

**Review by linasm — COMMENTED** [review]
Left a few nits, also I think index_test.go could use some coverage for `IsBootstrapped() = false` case. Also let's discuss the integration test offline, it feels like getting it right could be problematic.

File: src/dbnode/integration/cluster_remove_one_node_test.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	opts := NewTestOptions(t).
+		SetNamespaces([]namespace.Metadata{namesp}).
+		// Prevent snapshotting from happening too frequently to allow for the
+		// possibility of a snapshot occurring after the shard set is assigned,
+		// but not after the node finishes bootstrapping.
+		SetTickMinimumInterval(5 * time.Second)
+
+	minShard := uint32(0)
+	maxShard := uint32(opts.NumShards()) - uint32(1)
+	midShard := uint32((maxShard - minShard) / 2)
```
[the remark this answers is not in the document]
Comment (notbdu) [reply src/dbnode/integration/cluster_remove_one_node_test.go:71]: Will do, but just FYI the test setup has a default num shards and the integration tests I've seen generally just rely on the default options in the test setup. 

This test was mostly copied from an existing integration test.

[the remark this answers is not in the document]
Comment (notbdu) [reply src/dbnode/integration/cluster_remove_one_node_test.go:71]: https://github.com/m3db/m3/blob/master/src/dbnode/integration/options.go#L36-L91

File: src/dbnode/integration/cluster_remove_one_node_test.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+	setups, closeFn := NewDefaultBootstrappableTestSetups(t, opts, setupOpts)
+	defer closeFn()
+
+	// Write test data for first node.
+	topo, err := topoInit.Init()
+	require.NoError(t, err)
+	ids := []idShard{}
+
+	// Boilerplate code to find two ID's that hash to the first half of the
+	// shards, and one ID that hashes to the second half of the shards.
```
[the remark this answers is not in the document]
Comment (notbdu) [reply src/dbnode/integration/cluster_remove_one_node_test.go:124]: Mmm this is exactly the same code used in the other cluster add integration test.

**Review by linasm — APPROVED** [review]
LGTM

**Pull request merged**